### PR TITLE
Pawara

### PR DESCRIPTION
**Pull request by PawaraLiyanage**

### CODE REVIEW

**Review by kavicastelo — CHANGES_REQUESTED** [review]
change commented requests.
change endpoint is most important. because if you tring to call same endpoint in different methods, system will confuse and it going to crashed. 
you tring to call same endpoints like this
get -> /id
put -> /id
delete -> /id

in this attemt, imagine you request `get` method. but system identifies anathor two methods has same endpoint. so system tring to trigger both of those methods too. but it will failed because they are in different request methods. then system tring to send success request and two error requests at once. then system will going to crash.

File: src/main/java/com/hris/HRIS/controller/AttendanceController.java [hunk 1]
```diff
@@ -1,45 +1,67 @@
...
-        return attendanceModelOptional.map(ResponseEntity::ok).orElse(ResponseEntity.notFound().build());
+    // Endpoint to retrieve all attendance records
+    @GetMapping("/all")
+    public ResponseEntity<List<AttendanceModel>> getAllAttendance() {
+        List<AttendanceModel> allAttendance = attendanceService.getAllAttendance();
+        return new ResponseEntity<>(allAttendance, HttpStatus.OK);
     }
 
+    // Endpoint to retrieve a specific attendance record by ID
+    @GetMapping("/{id}")
```
[the remark this answers is not in the document]
Comment (PawaraLiyanage) [reply src/main/java/com/hris/HRIS/controller/AttendanceController.java:34]: Ok

File: src/main/java/com/hris/HRIS/controller/AttendanceController.java [hunk 2]
```diff
@@ -1,45 +1,67 @@
...
+        AttendanceModel attendanceModel = attendanceService.getAttendanceById(id);
+        if (attendanceModel != null) {
+            return new ResponseEntity<>(attendanceModel, HttpStatus.OK);
+        } else {
+            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+        }
+    }
 
+    // Endpoint to update an existing attendance record
+    @PutMapping("/{id}")
```
[the remark this answers is not in the document]
Comment (PawaraLiyanage) [reply src/main/java/com/hris/HRIS/controller/AttendanceController.java:45]: OK

File: src/main/java/com/hris/HRIS/dto/AttendanceDto.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+@Getter
+@Setter
+@ToString
+@AllArgsConstructor
+@NoArgsConstructor
+
+
+public class AttendanceDto {
+
+    private Long id;
```
Comment (kavicastelo) [inline src/main/java/com/hris/HRIS/dto/AttendanceDto.java:18]: id is a String

**Review by kavicastelo — CHANGES_REQUESTED** [review]
I see you are using uncommon returning method for this case. please check how others are handled to the returning values. this is not incorrect. this is correct but frontend developers might have a confusation when they start to implement your module. reason is you are directly returning the values to frontend without creating a json. some cases these kinds of requests data fetching not working properly or might be complicated.

> [!IMPORTANT]
> Test your APIs before submitting PR (Use POSTMAN). If you have any questions, contact me.

File: src/main/java/com/hris/HRIS/dto/AttendanceDto.java (hunk 3 above)
Comment (PawaraLiyanage) [reply src/main/java/com/hris/HRIS/dto/AttendanceDto.java:18]: Done

**Review by kavicastelo — APPROVED** [review]
Okay no worries.. If there is need a change I'll changed and updated

**Pull request merged**